public/app/features/query/components/QueryGroup.tsx

```diff
@@ -274,14 +280,35 @@ export class QueryGroup extends PureComponent<Props, State> {
             Data source
           </InlineFormLabel>
           <div className={styles.dataSourceRowItem}>
-            <DataSourcePicker
-              onChange={this.onChangeDataSource}
-              current={options.dataSource}
-              metrics={true}
-              mixed={true}
-              dashboard={true}
-              variables={true}
-            />
+            {config.featureToggles.drawerDataSourcePicker ? (
+              <DataSourcePickerWithHistory
+                onChange={this.onChangeDataSource}
+                current={options.dataSource}
+                metrics={true}
+                mixed={true}
+                dashboard={true}
+                variables={true}
+                fileUploadOptions={
+                  config.featureToggles.editPanelCSVDragAndDrop
+                    ? {
+                        onDrop: this.onFileDrop,
+                        maxSize: DFImport.maxFileSize,
+                        multiple: false,
+                        accept: DFImport.acceptedFiles,
+                      }
+                    : undefined
+                }
```

Comment:
For a more definitive props API, I might consider having `enableFileUpload: boolean` to enable or disable the D&D area, no matter what options you pass down.

Reply:
Fixed!